src/db/spanner/support.rs

```diff
@@ -26,10 +26,47 @@ use crate::{
 
 use super::{models::Result, pool::Conn};
 
-pub fn as_value(string_value: String) -> Value {
-    let mut value = Value::new();
-    value.set_string_value(string_value);
-    value
+pub trait ToSpannerValue {
+    fn to_spanner_value(&self) -> Value;
+}
+
+impl ToSpannerValue for String {
+    fn to_spanner_value(&self) -> Value {
+        let mut value = Value::new();
+        value.set_string_value(self.clone());
+        value
+    }
+}
+
+impl ToSpannerValue for i32 {
+    fn to_spanner_value(&self) -> Value {
+        let mut value = Value::new();
+        value.set_number_value(*self as f64);
+        value
+    }
+}
+
+impl ToSpannerValue for u32 {
+    fn to_spanner_value(&self) -> Value {
+        let mut value = Value::new();
+        value.set_number_value(*self as f64);
+        value
+    }
+}
+
+impl<T> ToSpannerValue for Vec<T>
```

Comment:
Same issue as above :( What I really wanted to do was:
```rust
impl<T, V> ToSpannerValue for T
where T: Iterator<Item=V>
      V: ToSpannerValue {
    ...
}
```
but it creates ambiguity for the compiler if someone "upstream" were to implement `Iterator<Item=V>` for `String`.